runtime/lib/mirrors_impl.dart

```diff
@@ -1665,7 +1667,10 @@ class _Mirrors {
     return classMirror;
   }
 
-  static TypeMirror reflectType(Type key) {
+  static TypeMirror reflectType(Type key, [List<Type> typeArguments]) {
+    if (typeArguments != null) {
+      key = _resolveType(key, typeArguments);
```

Comment:
This might conflict with @rmacnak-google comments on reflectType.
I think I like his idea by just instantiating generic type in this if statement and leaving everything else as it was before (sorry @rmacnak-google I didn't realize this in your very first comment about `_instantiateType`).

@floitschG would it be ok to go with `_instantiateType` approach for you?